src/app/content/search/utils.ts

```diff
@@ -175,3 +182,21 @@ export const findSearchResultHit = (
 ): SearchResultHit | undefined => {
   return results.find((result) => result.source.elementId === target.elementId);
 };
+
+export const createUpdatedNavigation = (targetPageId: string | null, book: Book) => {
```

Comment:
This function is not updating `navigation` so I think it should be maybe called something like `createRouteMatchOptions`
It should probably also take `book: Book, pageId: string` and return null if `page` was not found in a book.
This way it would be more generic. You could also declare the returned type so it is more clear what we expect